### PR TITLE
Add RSA-SHA2 support for the WinCNG backend

### PR DESCRIPTION
**Pull request by skundu07**
Added code to support RSA-SHA2 for WinCNG backend. It seems to work fine here. Let me know if this is okay.

### CODE REVIEW

**Comment by skundu07** [comment]
It seems there is some problem with "appveyor" test as it completed successfully for the first commit, and only some coding styles were changed later without modifying the actual code.

**Comment by mback2k** [comment]
I triggered a re-run to fix that issue.

File: src/wincng.c [hunk 1]
```diff
@@ -1209,12 +1227,24 @@ _libssh2_wincng_rsa_sha1_verify(libssh2_rsa_ctx *rsa,
...
-    return _libssh2_wincng_key_sha1_verify(rsa, sig, sig_len, m, m_len,
-                                           BCRYPT_PAD_PKCS1);
+   return _libssh2_wincng_key_sha_verify(rsa, SHA_DIGEST_LENGTH, sig, sig_len,
+                                         m, m_len, BCRYPT_PAD_PKCS1);
 }
 
 int
-_libssh2_wincng_rsa_sha1_sign(LIBSSH2_SESSION *session,
+_libssh2_wincng_rsa_sha2_verify(libssh2_rsa_ctx* rsa,
+   size_t hash_len,
```
Comment (mback2k) [inline src/wincng.c:1236]: The arguments should be indented like the first one, so in line with the opening brace `(`.

File: src/wincng.c [hunk 2]
```diff
@@ -1226,14 +1256,25 @@ _libssh2_wincng_rsa_sha1_sign(LIBSSH2_SESSION *session,
     unsigned long cbData, datalen, siglen;
     int ret;
 
+    if(hash_len == SHA_DIGEST_LENGTH)
+      paddingInfo.pszAlgId = BCRYPT_SHA1_ALGORITHM;
```
Comment (mback2k) [inline src/wincng.c:1260]: Mixed indentation levels between the if-statements. Should be 4 instead of 2 spaces here.

File: src/wincng.c [hunk 3]
```diff
@@ -2604,8 +2646,13 @@ _libssh2_supported_key_sign_algorithms(LIBSSH2_SESSION *session,
                                        size_t key_method_len)
 {
     (void)session;
-    (void)key_method;
-    (void)key_method_len;
+
+#if LIBSSH2_RSA_SHA2
+    if(key_method_len == 7 &&
+       memcmp(key_method, "ssh-rsa", key_method_len) == 0) {
```
Comment (mback2k) [inline src/wincng.c:2652]: Invalid indentation, should be 4 instead of 3 spaces.

**Review by mback2k — CHANGES_REQUESTED** [review]

File: src/wincng.c (hunk 1 above)
Comment (skundu07) [reply src/wincng.c]: Fixed it in the latest commit.

File: src/wincng.c (hunk 2 above)
Comment (skundu07) [reply src/wincng.c]: Fixed it in the latest commit.

File: src/wincng.c (hunk 3 above)
Comment (skundu07) [reply src/wincng.c]: Fixed it in the latest commit.

**Comment by willco007** [comment]
This looks good to me, @mback2k look good to you?

**Review by mback2k — APPROVED** [review]
Looks good to me, thanks!

**Pull request merged**